tests/eas/misfit.py

```diff
@@ -0,0 +1,400 @@
+        """
+        :returns: The absolute maximum load-balance interval in seconds
+        """
+        HZ = cls._get_hz(test_env)
+        return ((HZ * test_env.platform['cpus_count']) / 10) * (1. / HZ)
+
+    @classmethod
+    def _get_wload(cls, test_env):
+        raise NotImplemented()
+
+    @classmethod
+    def _getExperimentsConf(cls, test_env):
+        # Verify that the target has asymetric capacity
+        if not cls._has_asym_cpucapacity(test_env):
+            raise SkipTest(
+                'This test requires a target with asymetric CPU capacities. '
+                'SD_ASYM_CPUCAPACITY was not found.'
+            )
+
+        conf = {
+            'tag' : 'misfit',
+            'flags' : ['ftrace', 'freeze_userspace'],
+        }
+
+        if 'cpufreq' in test_env.target.modules:
+            available_govs = test_env.target.cpufreq.list_governors(0)
+            conf['cpufreq'] = {'governor' : 'performance'}
+
+        return {
+            'wloads' : cls._get_wload(test_env),
+            'confs' : [conf],
+        }
+
+class StaggeredFinishes(_MisfitMigrationBase):
+    """
+    Test Misfit task migration happens at idle balance (staggered test case)
+
+    This test spawns nr_cpus 100% tasks. The tasks running on bigger-capacity
+    CPUs will finish first, and it is expected of them to instantly pull the
+    tasks running on smaller-capacity CPUs via idle-balance.
+    """
+
+    # How long the tasks will be pinned to their "starting" CPU. Doesn't have
+    # to be long (we just have to ensure they spawn there), so arbitrary value
+    pin_delay = 0.001
+
+    # Somewhat arbitrary delay - long enough to ensure
+    # rq->avg_idle > sysctl_sched_migration_cost
+    idling_delay = 1
+
+    # How long do we allow misfit tasks to stay on LITTLE CPUs when big CPUs
+    # are free
+    allowed_misfit_pct = 10
+
+    @classmethod
+    def _get_wload(cls, test_env):
+        # rt-app json file is assembled by hand because we need per-phase
+        # CPU-affinity, and that's not (yet) feasible in LISA
+        wload = {
+            "global": {
+                "calibration": "__PVALUE__",
+                "default_policy": "SCHED_OTHER",
+                "duration": -1,
+                "logdir" : "__LOGDIR__"
+            },
+            "tasks": {}
+        }
+
+        task_template = {
+            "delay": int(cls.idling_delay * 1000000),
+            "loop": 1,
+            "phases": OrderedDict([("pinned", {}), ("free", {})]),
+            "policy": "SCHED_OTHER"
+        }
+
+        for key in task_template["phases"].iterkeys():
+            task_template["phases"][key]["loop"] = 1
+
+        task_template["phases"]["pinned"]["run"] = int(1e6 * cls.pin_delay)
+        # We're pinning stuff in the first phase, so give it ample time to
+        # clean the pinned logic out of balance_interval
+        task_template["phases"]["free"]["run"] = int(
+            1e6 * 1.1 * cls._get_max_lb_interval(test_env))
```

Comment:
Why 1.1?

Reply:
It's a very scientifically proven number to have something > 1 to ensure that even with the pinned logic we did get one balance attempt.